### ISSUE TITLE
UNION generates bad SQL for MySQL, when UNION subqueries contain duplicate column names from JOIN operations

### ISSUE THREAD

**Issue by javathought** [jOOQ/jOOQ#4856]
On a complex union with Mysql, I got this error Error :   

```
"Erreur SQL : jOOQ; bad SQL grammar [] nested exception is com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Duplicate column name ''"
```

The generated sql command is : 

``` sql
select * from (select col_a, col_b, ...
from table_a
left outer join table_b on ...
left outer join  table_c as hot_pce on ...
left outer join table_d on ...
left outer join table_e on ...
 ) x 
union
select * from (select col_a, col_b, ...
from table_abis
left outer join table_b on ...
left outer join  table_c as hot_pce on ...
left outer join table_d on ...
left outer join table_e on ...
 ) x 
```

The sql query is ok without `select * from (` and `x )`

The query is generated byt this code 

``` java
SelectOnConditionStep<Record19<String, String, String, String, 
                    String, String, String, Byte, String, 
                    String, String, String, Byte, String, 
                    String, String, String, Long, Long>>
                    query = dslContext.select(  )
            .from(TABLE_A)
                    .leftOuterJoin(TABLE_B).on()
                    .leftOuterJoin(TABLE_C).on()
                    .leftOuterJoin(TABLE_D).on()
                    .leftOuterJoin(TABLE_E).on();

SelectOnConditionStep<Record19<String, String, String, String,
                String, String, String, Byte, String,
                String, String, String, Byte, String,
                String, String, String, Long, Long>>
                query2 = dslContext.select(   )
            .from(TABLE_ABIS)
                    .leftOuterJoin(TABLE_B).on()
                    .leftOuterJoin(TABLE_C).on()
                    .leftOuterJoin(TABLE_D).on()
                    .leftOuterJoin(TABLE_E).on();

query.union(query2);

```

The errors occurs both on jOOQ 3.6.1 and 3.7.2

**Comment by lukaseder** [comment]
Thank you very much for reporting.

Indeed, in all databases, subqueries / derived tables are not allowed to have ambiguous / duplicate column names, while top level queries are allowed to do so.

In this case, the derived table is generated by jOOQ, so jOOQ should be able to handle this, internally - probably by renaming columns in the derived table, and then renaming them back again to the original names.

**Comment by lukaseder** [comment]
This should be solved by enumerating names for all subquery columns and restoring their names again in the top level query

**Comment by jsnorman** [comment]
how can i avoid the select *()x?  since in my environment , i need add a where for each select.

**Comment by jsnorman** [comment]
can the select*(...)x be replaced by select (...)x where 1=1?

**Comment by lukaseder** [comment]
@jsnorman: Thank you for your questions. Unfortunately, I'm not quite sure what you mean. Is this related to this issue here (duplicate column names in union subselects)? If not, would you mind opening a separate issue or directing your question to the jOOQ user group: https://groups.google.com/d/forum/jooq-user

Thank you very much

**Comment by nmeyvis** [comment]
Any updates on this? Is there a workaround for the latest version?

**Comment by lukaseder** [comment]
@nmeyvis: No, I'm afraid this issue still persists.

**Comment by lukaseder** [comment]
This issue seems to no longer persist. Our relevant integration tests doesn't fail in 3.13 and 3.12.4. Closing this as fixed.

**Comment by lukaseder** [comment]
Related recent fixes: #9751, #3676
